### PR TITLE
Test and log warning if checksum mismatched

### PR DESCRIPTION
**Pull request by o01eg**
Intended to be required in the forks.

### CODE REVIEW

**Comment by o01eg** [comment]
I also make clang build fallible because some update broke it.

File: test/system/ClientAppFixture.cpp [hunk 1]
```diff
@@ -15,6 +15,17 @@
 #include <boost/thread/thread.hpp>
 #include <boost/test/unit_test.hpp>
 
+namespace {
+    void AddOptions(OptionsDB& db) {
+#ifdef FREEORION_WIN32
+        db.Add<std::string>("misc.server-local-binary.path", UserStringNop("OPTIONS_DB_FREEORIOND_PATH"),        PathToString(GetBinDir() / "freeoriond.exe"));
```
Comment (geoffthemedio) [inline test/system/ClientAppFixture.cpp]: should / could this be in ClientNetworking instead of being duplicated here and chmain ?

**Comment by geoffthemedio** [comment]
otherwise no comment...

File: test/system/ClientAppFixture.cpp (hunk 1 above)
Comment (o01eg) [reply test/system/ClientAppFixture.cpp]: AI client doesn't need it.

Comment (geoffthemedio) [reply test/system/ClientAppFixture.cpp]: having it be available when not needed is probably better than the code duplication, I suppose?

**Comment by geoffthemedio** [comment]
I already responded with no comment... Do you have a specific question?

File: test/system/ClientAppFixture.cpp (hunk 1 above)
Comment (o01eg) [reply test/system/ClientAppFixture.cpp]: I need to call it before `InitDirs` and test fixtures call `InitDirs` after `ClientApp` constructed so I have to properly order calls to move it into ClientNetworking

Comment (geoffthemedio) [reply test/system/ClientAppFixture.cpp]: OK

**Comment by o01eg** [comment]
Suddenly it gets "stack overflow" error on Windows build. Could be related to options db refactoring.

**Comment by o01eg** [comment]
Locally (gentoo amd64) I get:
 ```
5: 13:47:10.706615 {0x00007f55a1942740} [debug] test : NamedValueRefManager.cpp:116 : NamedValueRefManager partial checksum: 0
5: 13:47:10.706628 {0x00007f55a1942740} [debug] test : NamedValueRefManager.cpp:119 : NamedValueRefManager second partial checksum: 30563
5: 13:47:10.706862 {0x00007f55a1942740} [debug] test : NamedValueRefManager.cpp:123 : NamedValueRefManager checksum: 1045825
5: /mnt/another/srcs/GIT/freeorion/test/system/TestChecksum.cpp:10: error: warning: in "TestChecksum/compare_checksum": condition checksums["NamedValueRefManager"] == 1042814 is not satisfied [1045825 != 1042814]
```

CI (ubuntu amd64 18.04 and 20.04) gets:
```
5: 10:33:14.365178 {0x00007f78c2cdb740} [debug] test : NamedValueRefManager.cpp:116 : NamedValueRefManager partial checksum: 0
5: 10:33:14.365201 {0x00007f78c2cdb740} [debug] test : NamedValueRefManager.cpp:119 : NamedValueRefManager second partial checksum: 30563
5: 10:33:14.365601 {0x00007f78c2cdb740} [debug] test : NamedValueRefManager.cpp:123 : NamedValueRefManager checksum: 1042814
5: /home/runner/work/freeorion/freeorion/test/system/TestChecksum.cpp:10: error: info: check checksums["NamedValueRefManager"] == 1042814 has passed
```

Windows CI:
```
  5: D:/a/freeorion/freeorion/test/system/TestChecksum.cpp:10: error: info: check checksums["NamedValueRefManager"] == 1042814 has passed
```

**Pull request merged**